tests/platform.lua

```diff
@@ -0,0 +1,34 @@
+-- Determine some platform information used for tests
+-- Adapted from: https://github.com/keplerproject/luarocks/blob/master/src/luarocks/cfg.lua
+local platform = {}
+local system = io.popen("uname -s"):read("*l")
+local proc = io.popen("uname -m"):read("*l")
+
+if proc:match("i[%d]86") then
+  platform.proc = "x86"
+elseif proc:match("amd64") or proc:match("x86_64") then
+  platform.proc = "x86_64"
+else
+  platform.proc = nil
+end
+
+if system == "Darwin" then
+  platform.unix = true
+  platform.macosx = true
+elseif system == "Linux" then
+  platform.unix = true
+  platform.linux = true
+elseif system and system:match("^CYGWIN") then
+  platform.unix = true
+  platform.cygwin = true
+elseif system and system:match("^Windows") then
```

Comment:
Yeah, I thought the same thing but don't have a Windows box to verify. These checks are taken from the luarocks cfg module. Cygwin and Mingw at least expose uname and these seem to check out for those environments.

I suppose it can be adjusted to use `ver` or `winver` or some other check for Windows (systeminfo?) but I don't have a Windows box to test it against.